Xamarin.Forms.Platform.Android/AppCompat/SwitchRenderer.cs

```diff
@@ -81,6 +87,36 @@ protected override void OnElementChanged(ElementChangedEventArgs<Switch> e)
 
 				e.NewElement.Toggled += HandleToggled;
 				Control.Checked = e.NewElement.IsToggled;
+				defaultOnColor = Control.TrackTintList;
+				defaultMode = Control.TrackTintMode;
+				UpdateOnColor();
+			}
+		}
+
+		protected override void OnElementPropertyChanged(object sender, PropertyChangedEventArgs e)
+		{
+			if (e.PropertyName == Switch.OnColorProperty.PropertyName)
+				UpdateOnColor();
+		}
+
+		private void UpdateOnColor()
```

Comment:
We do not use the private keyword as it is the default accessibility level in C#.